src/Promitor.Agents.Scraper/Validation/MetricDefinitions/ResourceTypes/IotHubMetricValidator.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using GuardNet;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
+using Promitor.Core.Scraping.Configuration.Model.Metrics.ResourceTypes;
+using Promitor.Agents.Scraper.Validation.MetricDefinitions.Interfaces;
+
+namespace Promitor.Agents.Scraper.Validation.MetricDefinitions.ResourceTypes
+{
+    internal class IotHubMetricValidator : IMetricValidator
+    {
+        public IEnumerable<string> Validate(MetricDefinition metricDefinition)
+        {
+            Guard.NotNull(metricDefinition, nameof(metricDefinition));
+
+            foreach (var resourceDefinition in metricDefinition.Resources.Cast<IotHubResourceDefinition>())
+            {
+                if (string.IsNullOrWhiteSpace(resourceDefinition.IotHubName))
```

Comment:
More of a general nit - Can we rename everything with `Iot` to `IoT`? Sorry :/

Reply:
done - was a bit nasty because file names. Was thinking about that first, since its the correct casing, then decided against, for readability. But if you are in favor of IoT, lets go that way.